src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -150,6 +150,33 @@ impl<'a, 'mir, 'tcx, M: Machine<'a, 'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M>
                 }
                 self.write_scalar(val, dest)?;
             }
+            "rotate_left" | "rotate_right" => {
```

Comment:
cc @oli-obk @RalfJung is it necessary to "emulate" the operation here? Can we simply call the stdlib’s rotation function instead? Easily?

Reply:
Yes, you should be able to do it just like with the calls above to `numeric_intrinsic`

Reply:
I've pushed a new implementation that still does the emulation, but directly on the bit representation. I'm not sure how I can base this directly on the libcore methods, as they are integer-size specific.

Reply:
Yeah, this seems fine to me. What do you think @oli-obk?

Alternative would be to do a `match width_bits` and call the libcore method on correspondingly cast value.